pkg/scalers/azure_monitor_scaler.go

```diff
@@ -148,6 +149,24 @@ func parseAzureMonitorMetadata(config *ScalerConfig) (*azureMonitorMetadata, err
 
 	meta.scalerIndex = config.ScalerIndex
 
+	azureResourceManagerEndpointProvider := func(env az.Environment) (string, error) {
+		return env.ResourceManagerEndpoint, nil
+	}
+	azureResourceManagerEndpoint, err := azure.ParseEnvironmentProperty(config.TriggerMetadata, "azureResourceManagerEndpoint", azureResourceManagerEndpointProvider)
+	if err != nil {
+		return nil, err
+	}
+	meta.azureMonitorInfo.AzureResourceManagerEndpoint = azureResourceManagerEndpoint
+
+	activeDirectoryEndpointProvider := func(env az.Environment) (string, error) {
```

Comment:
Should we extract this function and place in azure package? I can see that we are using a similar provider in other azure scalers

Reply:
oh, you are right, all the PRs are already opened... yes, after merging all the PRs we can do it in another separate PR